src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -445,6 +445,28 @@ struct ConvertPrecision<std::tuple<ov::float16, dst_t>> {
     }
 };
 
+template<>
+struct ConvertPrecision<std::tuple<ov::float16, float>> {
+    void operator()(ConvertContext & ctx) {
+        auto src = static_cast<const ov::float16 *>(ctx.srcPtr);
+        auto dst = static_cast<float *>(ctx.dstPtr);
+
+        if (ctx.interimPrc.is_real()) {
+            parallel_for(ctx.size, [&](size_t i) {
+                dst[i] = static_cast<float>(src[i]);
```

Comment:
Given the  implementation of `ov::float16` data type, the compiler will most likely use scalar instructions to preform data conversions in this scenario, which eventually ruins performance.
Please take a closer look at the `ConvertPrecision<std::tuple<ov::float16, dst_t>>` template instance. There a special jit kernel is used that leverages F16C sse extension, which provides special HW acceleration to convert fp16<->fp32.
Also, given the fact that fp16 data type has lowest lover/upper bound absolute value, it seems to be safe to remove clamping for any intermediate floating point types when fp16 data type is either src or dst.
What do you think?

Reply:
> Given the implementation of `ov::float16` data type, the compiler will most likely use scalar instructions to preform data conversions in this scenario, which eventually ruins performance.

Oops, true, there are no cvtph intrinsics for bfloat16, that's why it is that straightforward, I should have thought about it before, sorry. Will make it using jit_convert. I should also use batching as in other specializations I guess. 



> Also, given the fact that fp16 data type has lowest lover/upper bound absolute value, it seems to be safe to remove clamping for any intermediate floating point types when fp16 data type is either src or dst.

Also, true, will do! Thanks

Reply:
Ah, ok probably makes not much sense, since we still need the values not the comparison result

Reply:
It's hard to say for all the compilers, but at least gcc vectorize the clamp loop at least for the floating data types.